src/ProductConstructionService/ProductConstructionService.Api/AuthenticationConfiguration.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Maestro.Authentication;
+using Microsoft.DotNet.GitHub.Authentication;
+
+namespace ProductConstructionService.Api;
+
+public static class AuthenticationConfiguration
+{
+    public const string GitHubAuthenticationKey = "GitHubAuthentication";
```

Comment:
Maybe we should just bind the section to a named strong-typed class?
E.g. I see this is being passed down and bound to .Microsoft.DotNet.Web.Authentication.GitHub.GitHubAuthenticationOptions` eventually, so maybe we can speed this up?

I am not fond of pulling configuration out key by key when you could have a full class that gives you the schema for your config right away.

Reply:
Maybe, if you're changing the DI in `dnceng-shared`, you could add another overload for the `AddGitHubAuth()` and allow `GitHubAuthenticationOptions` to be passed instead of this `ISection`

Reply:
yeah, why not, we'll have a few less consts, I'll add the change

Reply:
I tried doing this and running locally, but something wasn't properly binding in the Auth classes when registering `GitHubAuthenticationOptions` like this. Perhaps it was the `Option` name parameter we were talking about

Reply:
Ok, unfortunate but we can revisit later

Reply:
I'll create an issue and put a TODO so we don't forget